### PR TITLE
Add CIBA grant type to oidc-client

### LINKED ISSUE #34356: Add CIBA grant type to oidc-client
### Description

Add `CIBA` grant type: `urn:openid:params:grant-type:ciba` to `OidcClientConfig`, as some providers already support CIBA protocol  (e.g. Keycloak).

[OpenID Connect Client-Initiated Backchannel Authentication Flow - Core 1.0](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html)

https://github.com/quarkusio/quarkus/blob/1eb84375c38d9619553f6b37a07f85745e92d80f/extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/OidcClientConfig.java#L54-L88

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by Eng-Fouad**
Fixes #34356

### CODE REVIEW

**Review by sberyozkin — APPROVED** [review]

**Comment by sberyozkin** [comment]
Sorry @Eng-Fouad, I thought we should discuss an option of testing it given that all other grant types are tested AFAIK.
Can you have look please at `integration-tests/oidc-client-wiremock` and see can be done ?

**Comment by Eng-Fouad** [comment]
> 

I have no much time atm. I will try to take a look on it next week.

**Comment by sberyozkin** [comment]
@Eng-Fouad Sure, give a try please, I appreciate you opening a PR, I'll try to help with the test if you'll find it not easy to set up, but it should be mostly a copy and paste exercise - copy one of the wiremock stubs there and create the one for the CIBA grant, add one more oidc client tenant config with this grant and copy some test and use this tenant config.

**Comment by Eng-Fouad** [comment]
Does `oidc-client` currently support sending request to `/auth` endpoint? Because `CIBA` requires to call [`Backchannel Authentication Endpoint`](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#auth_backchannel_endpoint) before calling `/token` endpoint.

**Comment by Eng-Fouad** [comment]
@sberyozkin Added test cases.

File: integration-tests/oidc-client-wiremock/src/test/java/io/quarkus/it/keycloak/OidcClientTest.java [hunk 1]
```diff
@@ -28,6 +30,9 @@
 @QuarkusTestResource(KeycloakRealmResourceManager.class)
 public class OidcClientTest {
 
+    @InjectWireMock
```
Comment (sberyozkin) [inline integration-tests/oidc-client-wiremock/src/test/java/io/quarkus/it/keycloak/OidcClientTest.java:33]: @Eng-Fouad This injection is not necessary, `KeycloakRealmResourceManager` is starting a Wiremock server and you have stubs created there

Comment (Eng-Fouad) [reply integration-tests/oidc-client-wiremock/src/test/java/io/quarkus/it/keycloak/OidcClientTest.java:33]: Then how can I access `WireMockServer` instance? I need it to invoke `server.setScenarioState()`.

Comment (sberyozkin) [reply integration-tests/oidc-client-wiremock/src/test/java/io/quarkus/it/keycloak/OidcClientTest.java:33]: Oh, I see you you using the scenario...

**Comment by sberyozkin** [comment]
@Eng-Fouad Thanks, the test looks OK, have a look at
https://github.com/quarkusio/quarkus/blob/main/CONTRIBUTING.md#build

It should be very easy to build it, and then `mvn clean install -Dtest=OidcClientTest#testCibaGrant` in `integration-tests/oidc-client-wiremock`

File: integration-tests/oidc-client-wiremock/src/test/java/io/quarkus/it/keycloak/KeycloakRealmResourceManager.java [hunk 2]
```diff
@@ -67,6 +67,45 @@ public Map<String, String> start() {
                         .withBody(
                                 "{\"access_token\":\"temp_access_token\", \"expires_in\":4}")));
 
+        server.stubFor(WireMock.post("/ciba-token")
+                .withRequestBody(matching(
+                        "grant_type=urn:openid:params:grant-type:ciba&client_id=quarkus-app&client_secret=secret&auth_req_id=16cdaa49-9591-4b63-b188-703fa3b25031"))
```
Comment (Eng-Fouad) [inline integration-tests/oidc-client-wiremock/src/test/java/io/quarkus/it/keycloak/KeycloakRealmResourceManager.java:72]: @sberyozkin For some reason, `client_id` and `client_secret` are not sent from `oidc-client`. If I remove them from here, this test passes.

Comment (Eng-Fouad) [reply integration-tests/oidc-client-wiremock/src/test/java/io/quarkus/it/keycloak/KeycloakRealmResourceManager.java:72]: I see now. It sends it as BASIC auth.

**Comment by Eng-Fouad** [comment]
@sberyozkin All tests should pass now.

**Review by sberyozkin — APPROVED** [review]
Great effort @Eng-Fouad, thanks

**Pull request merged**